homeassistant/components/myq/__init__.py

```diff
@@ -63,3 +73,42 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+class MyQEntity(CoordinatorEntity):
+    """Base class for MyQ Entities."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, device) -> None:
+        """Initialize class."""
+        super().__init__(coordinator)
+        self._device = device
+        self._attr_unique_id = device.device_id
+
+    @property
+    def name(self):
+        """Return the name if any, name can change if user changes it within MyQ."""
+        return self._device.name
+
+    @property
+    def device_info(self):
```

Comment:
This can move to the init as well

Reply:
Reason I kept it where it was is because some of the information can change through updates (i.e. name, firmware version).